pytest/tests/sanity/backward_compatible.py

```diff
@@ -1,49 +1,71 @@
 #!/usr/bin/env python
 """
-This script runs node from stable branch and from current branch and makes
-sure they are backward compatible.
+This script runs node from two specified branches and makes sure they are compatible.
+First the node using boot_branch starts and the node using target_branch should boot
+from the former.
+
+The branches to use can be specified as arguments to this script.
+By default boot_branch is rc and target_branch is the current_branch.
+
+./backward_compatible.py <boot_branch> <target_branch>
+
+Both <boot_branch> and <target_branch> must be either the name of an existing branch or:
+- rc: latest release candidate (used for testnet)
+- beta: latest beta (used for betanet)
+- stable: latest branch (used for mainnet)
+
+To create genesis and required configuration to start a network, the binary compiled with boot_branch is used,
+but a particular branch can be specified with the third argument:
+
+./backward_compatible.py <boot_branch> <target_branch> <config_branch>
```

Comment:
what's the use case of having `config_branch`?

Reply:
When the config files are initialized with the newer node, usually the old node might not work at all.

Reply:
So the config_branch is more useful, when the backward_compatible test is used in reverese, i.e. we are booting from the node with newer version, in this case it is possible to initialize configs with the older version, by passing config_branch=target_branch

Reply:
but what guarantees that it will work? What happens when the node in the newer version cannot read the config from the old version?

Reply:
The case where older node can't read older version is more common as we add more configs than we remove, and even when we remove, I think the old node will correctly parse the config and add the default value for missing fields.

When the config generated by both versions is incompatible to the other version, that would require to update this add some migration from configs in this test, but I don't think it is necessary. 

As mentioned before the main motivation to add the third argument is to be able to boot from the newer node, but using default configs generated by the older node!